make date-check more lightweight (#1394)
rust-lang/rustc-dev-guide: src/contributing.md

    the project.

  - The date the comment was added, e.g. instead of writing _"Currently, ..."_
    or _"As of now, ..."_,
    consider adding the date, in one of the following formats:
    - Jan 2021
    - January 2021
    - jan 2021
    - january 2021

>     There is a CI action (in "~/.github/workflows/date-check.yml")

Reviewer: ```suggestion
    There is a CI action (in `~/.github/workflows/date-check.yml`)
```

    that generates a monthly issue with any of these that are over 6 months old.

    For the action to pick the date, add this annotation:

        <!-- date-check -->

    Example:

        As of <!-- date-check --> Jul 2022, the foo did the bar.

    For cases where the date should not be part of the visible rendered output,
    use the following instead:

>         <!-- date-check: Jul 2022 -->

Reviewer: nit: could you use fenced code blocks and mark it as markdown?

  - The date the comment was added, e.g. instead of writing _"Currently, ..."_
    or _"As of now, ..."_, consider writing
    _"As of January 2021, ..."_.

>     Try to format the date as `<MONTH> <YEAR>` to ease search.

Reviewer: I'd rather keep this line (or something like it) since the format does matter.

>     We have CI action (in `"~/.github/workflows/date-check.yml"`)
>     that generates an issue if any of these are over 6 months old.

Reviewer: ```suggestion
    We have a CI action (in `~/.github/workflows/date-check.yml`)
    that generates a monthly issue with any of these that are over 6 months old.
```
